lib/ldapUser.class.php

```diff
@@ -113,20 +115,35 @@ protected function getLdapConnection()
 
             $this->ldapConnection = $connection;
 
-            return $connection;
+            return $this->ldapConnection;
         }
     }
 
     protected function ldapBind($username, $password)
     {
+        $tls_encryption = sfConfig::get('app_ldap_enable_tls_encryption', true);
         if ($conn = $this->getLdapConnection()) {
             $base_dn = (string) QubitSetting::getByName('ldapBaseDn');
             $bind_attribute = (string) QubitSetting::getByName('ldapBindAttribute');
             $dn = $bind_attribute.'='.$username.','.$base_dn;
 
+            // Close LDAP connection if TLS encryption failed to start
+            if ($tls_encryption && false == ldap_start_tls($conn)) {
+                $this->logger->err('Error starting TLS encryption for LDAP authentication.');
+                ldap_close($conn);
+
+                return false;
+            }
+            if (!$tls_encryption) {
```

Comment:
just to bring to your attention that this line has changed from the gitlab commit since it fails php fixer test